### DISCUSSION TITLE
Remove dashes and paste

### DISCUSSION THREAD

**Discussion by slvrstn** [hluk/CopyQ#1765] [Q&A]
I couldn't follow the documentation for adding a command so I'd appreciate some help in what I think should be relatively easy.
I want a command that will paste the most recently copied item after replacing all dashes with spaces.
So, if the last thing I copied to clipboard was
`Remove-all-of-the-dashes`
it would paste as
`Remove all of the dashes`

Appreciate any help with the config needed in copyq to do this. Thanks!

**Comment by slvrstn** [comment]
OK, I think I got it ...
```
copyq:
var text = str(clipboard())
var newText = text.replace(/-/g, ' ')
if (text == newText)
  abort();

copy(newText)
paste()
```

I didn't really find documentation for `replace`, but I did find an example with the regex version
(which got me the global replace option)
Thanks for rubber-ducking!

Oh, and learning that the language is basically Javascript helps for discovery.
Now that I know I guess I can't *not* see it in the docs.

**Reply by hluk** [reply]
I can see what you mean. I think there were more mentions about JavaScript in the docs in the past but somehow got lost after many updates.

I've updated the docs and mention JavaScript also here: https://copyq.readthedocs.io/en/latest/scripting-api.html

**Comment by elmaure** [comment]
I trying so hard to do something similiar. I want to remove dots from a string, for example, 34.803.198 when i copy i want to get 34803198. I wrote this, but does not work.

copyq:
vartext=str(clipboard())
varnewText=text.replace(/./g,'')
if(text==newText)
abort();

copy(newText)
paste()

**Reply by elmaure** [reply]
I also tried 

copyq:
var text = str(input())
if ( text.indexOf(' ') == -1 )
    abort()
    
var newText = text.replace(/ /g, '')
setData(mimeText, newText)
removeData(mimeHtml)
copy(newText)

**Reply by hluk** [reply]
You need to escape the dot in the regular expression (otherwise it has special meaning: "match any character"):

```js
varnewText=text.replace(/\./g,'')
```

**Reply by elmaure** [reply]
Thanks you Lukas

**Comment by Saccadio** [comment]
I am trying again something similar, but I have several problems with it. I want to paste plain text where all NL LF characters are replaced with a space character. So, this is (one of) my tries:

copyq:
copy()
var text = input()

if (text) {
  var newText = text.replace(/[\r\n]+/g,' ')
  copy(newText)  
  paste()
}

but I get a notification that property  "replace "of scriptable byte array is not a function. I am NOT proficient in regex expressions, nor Javascript, but this I do not understand.

Before this error, CopyQ complained about the last right parentheses, showing it in green together with the following text "copy (new...", thus  seemed to parse something wrong. So, I copied the "text.replace..." (which was copied from CopyQ's code window back into CopyQ - which surprisingly fixed the problem, so the command ran without errors, but did not replace with space. So, I tried some other variations - and are now back with the one above (that worked) - BUT now I get the error again????? 

What is it that I do not understand about how to enter the code - and what is it that I am not understanding about the regex, so that it does not replace with blanks. The pattern seems to work on Regex101.com

Btw, this is included in many copy managers, so that it is easy to paste text into e.g. subject fields in emails or filenames. Any thoughts on including it as standard in CopyQ (which I really like)

**Reply by hluk** [reply]
Try converting the ByteArray object (returned by `input()`) to string first: `var text = str(input())`

**Reply by Saccadio** [reply]
Thanks. Now I understand the error message. The code now reads:
copyq:
copy()
var text = str(input())

if (text) {
  var newText = text.replace(/[\r\n]+/g,' ' )
  copy (newText)  
  paste()
}

I do not get an error message, but I also do not get anything pasted? When I first altered according to your suggestion, I got an error message that it could not paste.

In the log it reads:
Warning [2023-02-01 12:42:00.128] <monitorClipboard-35288>: [qt.qpa.mime] QtWarning: Retrying to obtain clipboard.

**Reply by Saccadio** [reply]
Now I have tried and tried again and do not get the gist of this.

I even tried this, which should be a nobrainer:

copyq:
copy()
var text =str( input())  

  var newText = text
  copy (newText)  
  paste()

But also here - nothing is pasted. Why?

**Reply by hluk** [reply]
Unfortunately, `paste()` may not work on some systems.

See also: [Why does pasting from CopyQ not work?](https://copyq.readthedocs.io/en/latest/faq.html#why-does-pasting-from-copyq-not-work)

**Reply by Saccadio** [reply]
Thanks for your reply. At least I may not be crazy.

So, CopyQ paste does not work on Windows 11? The paste settings suggested in your link were set already.

If this is the case for all users of W11, I would humbly suggest that you implement the feature I am trying to implement as standard - i..e any consecutive sequence of characters not allowed in filename (/t /r /l  - possibly : and other special characters that I do not remember off hand) are removed and replaced by one space character.

Why does Paste not work? I am just trying to copy and paste in a Notepad window. Nothing fancy
